src/Microsoft.OData.Client/ClientEdmModel.cs

```diff
@@ -106,13 +107,12 @@ public IEdmEntityContainer EntityContainer
         }
 
         /// <summary>
-        /// Gets the state of whether the edm structured schema elements have been set.
+        /// Gets the state of edm structured schema elements keyed by their Name.
         /// </summary>
-        internal List<IEdmSchemaElement> EdmStructuredSchemaElements
+        internal ConcurrentDictionary<string, IEdmSchemaElement> EdmStructuredSchemaElements
         {
-            get;
-            set;
-        }
+            get; 
+        } = new ConcurrentDictionary<string, IEdmSchemaElement>();

```

Comment:
the property_initializer syntax does not work in the VS2013 compiler, which we use for official builds. You'll need to change this.